distribution/sc-proxy-dex/src/proxy_common.rs

```diff
@@ -31,6 +36,78 @@ pub trait ProxyCommonModule {
         Ok(())
     }
 
+    #[payable("*")]
+    #[endpoint]
+    fn acceptPay(
+        &self,
+        #[payment_token] token_id: TokenIdentifier,
+        #[payment] amount: Self::BigUint,
+    ) {
+        if self.current_tx_accepted_funds().len() > MAX_FUNDS_EXTRIES {
```

Comment:
why is this condition needed ?

Reply:
because invalidation is only done in enter/exit/claimRewards endpoint, one might send funds just to increase the storage indefinitely